include/exadg/operators/inverse_mass_operator.h

```diff
@@ -91,9 +109,36 @@ class InverseMassOperator
     }
   }
 
+  void
+  initialize_inverse_mass_operator_with_block_jacobi()
+  {
+    // initialize mass operator
+    dealii::AffineConstraints<Number> const & constraint =
+      matrix_free->get_affine_constraints(dof_index);
+
+    MassOperatorData<dim> mass_operator_data;
+    mass_operator_data.dof_index  = dof_index;
+    mass_operator_data.quad_index = quad_index;
+
+    mass_operator.initialize(*matrix_free, constraint, mass_operator_data);
+
+    // build a BlockJacobiPreconditioner and use the vmult(dst,src) for applying the inverse mass
+    // operator on  source the vector
+    mass_preconditioner =
+      std::make_shared<BlockJacobiPreconditioner<MassOperator<dim, n_components, Number>>>(
+        mass_operator);
+  }
+
   dealii::MatrixFree<dim, Number> const * matrix_free;
 
   unsigned int dof_index, quad_index;
+
+  // BlockJacobi Preconditoner to be used when the MatrixFreeInverseMass is not available
+  bool explicit_matrix_free_inverse_mass_available;
+
+  MassOperator<dim, n_components, Number> mass_operator;
+
+  std::shared_ptr<PreconditionerBase<Number>> mass_preconditioner;
```

Comment:
Here, you could use directly `BlockJacobiPreconditioner` instead of `PreconditionerBase` for clarity. We do not need dynamic polymorphism in this case.